### PR TITLE
Dock lights

### PR DESCRIPTION
**Pull request by Kurfursten**
triggered
https://cdn.discordapp.com/attachments/239823463802470400/417192374859464714/2018-02-24_23-32-15.webm

Docking lights. If you're a madman you can use the multitool to unlink them or change their link id, but for now this serves no purpose other than making things non-OSHA compliant. Perhaps some day you will be able to build these.

MAPPERS: id_tag should match the areaname var of the docking_port/destination you've set these up next to, (e.g.: "main research department").

🆑 
* rscadd: Added docking lights.

### CODE REVIEW

**Comment by FudgePucker** [comment]
Maybe make a soldering iron board and a frame out of ref glass as the recipe for making them? :+1:

**Comment by FudgePucker** [comment]
Wait I just remembered glass might not be the best because it doesn't have much of crafting options. Make the frame out of metal, use a sheet of ref glass then insert the circuit.

File: code/game/objects/structures/docking_port.dm [hunk 1]
```diff
@@ -232,3 +245,35 @@ var/global/list/all_docking_ports = list()
...
+	icon_state = "docklight"
+	light_color = LIGHT_COLOR_ORANGE
+	machine_flags = MULTITOOL_MENU
+	var/triggered = 0
+	var/id_tag = "" //Mappers: This should match the areaname of the target destination port.
+	//Examples: "main research department", "research outpost", "deep space", "station auxillary docking", "north of the station", etc.
+
+/obj/machinery/docklight/New()
+	..()
+	dockinglights += src
```
Comment (DamianX) [inline code/game/objects/structures/docking_port.dm:264]: Do the opposite of this inside `Destroy()`

**Review by DamianX — CHANGES_REQUESTED** [review]

**Comment by Shadowmech88** [comment]
Not that I necessarily think these shouldn't be added, but the airlocks which are adjacent to docking ports do already have flashing warning lights when there is an incoming or departing ship.

**Comment by Probe1** [comment]
@Shadowmech88 Yea the discussion/PR came from larger ships being too big to have the airlock in view.  So you get plowed without warning.

File: code/game/objects/structures/docking_port.dm (hunk 1 above)
Comment (Kurfursten) [reply code/game/objects/structures/docking_port.dm:264]: Went ahead and committed this via the web app

**Review by DamianX — APPROVED** [review]

**Pull request merged**